docs/docker/building-net-docker-images.md

```diff
@@ -0,0 +1,189 @@
+
+```console
+$ yo docker
+
+     _-----_     ╭──────────────────────────╮
+    |       |    │   Welcome to the Docker  │
+    |--(o)--|    │        generator!        │
+   `---------´   │     Let's add Docker     │
+    ( _´U`_ )    │  container magic to your │
+    /___A___\   /│           app!           │
+     |  ~  |     ╰──────────────────────────╯
+   __'.___.'__
+ ´   `  |° ´ Y `
+
+? What language is your project using? (Use arrow keys)
+❯ .NET Core
+  Golang
+  Node.js
+
+```
+
+- Select `.NET Core` as the project type
+- `rtm` for the version of .NET Core
+- `Y` the project uses a web server
+- `5000` is the port the Web API application is listening on (i.e. http://localhost:5000)
+- `api` for the image name
+- `api` for the service name
+- `api` for the compose project 
+- `Y` to overwrite the current Dockerfile
+
+When the generator is complete the following files are added to the project
+
+- .vscode/launch.json
+- Dockerfile.debug
+- Dockerfile
+- docker-compose.debug.yml
+- docker-compose.yml
+- dockerTask.ps1
+- dockerTask.sh
+- .vscode/tasks.json
+
+The generator creates a two Dockerfiles.
+
+**Dockerfile.debug** - this file is based on the **microsoft/dotnet:1.0.0-preview2-sdk** image which if you note from the list of image variants, includes the SDK, CLI and .NET Core. Aptly named ".debug", will be the image used for debugging and development (F5) thus producing a larger image roughly 540MB.
+
+**Dockerfile** - this image is the release image based on **microsoft/dotnet:1.0.0-core** and should be used for production. This image when built is approximately 253 MB.
+
+### Creating the Docker images
+Using the `dockerTask.sh` script we can build or compose the image and container for the **api** application for a specific environment. Build the **debug** image by running the following command.
+
+```bash
+./dockerTask.sh build debug
+```
+
+The image will build the ASP.NET application, run `dotnet restore`, add the debugger to the image, set an `ENTRYPOINT` and finally copy the app to the image. The result is a Docker image named *api* with a `TAG` of *debug*.  See the images on the machine using `docker images`.
+
+```bash
+docker images
+
+REPOSITORY          TAG                  IMAGE ID            CREATED             SIZE
+api                 debug                70e89fbc5dbe        a few seconds ago   779.8 MB
+```
+
+Another way to generate the image and run the application within the Docker container is to open the application in Visual Studio Code and use the debugging tools. 
+
+Select the debugging icon in the View Bar on the left side of VS Code.
+
+![vscode debugging icon](../images/debugging_debugicon.png)
```

Comment:
the image files should be inside a folder called media and then inside a folder with the same name as the topic
so it would be ./media/building-net-docker-images/debugging_debugicon.png